pulsar-functions/instance/src/main/python/python_instance_main.py

```diff
@@ -102,6 +102,22 @@ def main():
       zpfile = zipfile.ZipFile(str(args.py), 'r')
       zpfile.extractall(os.path.dirname(str(args.py)))
     sys.path.insert(0, os.path.dirname(str(args.py)))
+  elif os.path.splitext(str(args.py))[1] == '.zip':
+    # Assumig zip file with format func.zip
+    # extract to folder function
+    # internal dir format
+    # "func/src"
+    # "func/requirements.txt"
+    # "func/deps"
+    # run pip install to target folder  deps folder
+    zpfile = zipfile.ZipFile(str(args.py), 'r')
+    zpfile.extractall(os.path.dirname(str(args.py)))
+    basename = os.path.splitext(str(args.py))[0]
+    requirements_txt_file = os.path.join(os.path.dirname(str(args.py)), basename, "requirements.txt")
+    deps_file = os.path.join(os.path.dirname(str(args.py)), basename, "deps")
+    cmd = "pip install -t %s -r %s --no-index --find-links %s" % (os.path.dirname(str(args.py)), requirements_txt_file, deps_file)
+    os.system(cmd)
```

Comment:
We should fail here and print some error message if the pip install didn't succeed

Reply:
Done